ha/core/event_manager/subscribe_event.py

```diff
@@ -31,9 +33,25 @@ def __init__(self, resource_type : RESOURCE_TYPES, states : List[RESOURCE_STATUS
             states (list): States
         """
         self.states = []
+        self.functional_types = []
         self.resource_type = resource_type.value \
             if isinstance(resource_type, RESOURCE_TYPES) \
                 else resource_type
         for state in states:
             st = state.value if isinstance(state, RESOURCE_STATUS) else state
             self.states.append(st)
+
+        try:
+            resource_func_types = FUNCTIONAL_TYPES[self.resource_type.upper()]
+        except KeyError as err:
+            raise Exception(f"{err} is not a valid FUNCTIONAL_TYPE of {self.resource_type}")
+
+        if functional_types:
+            for func_type in functional_types:
+                func_type = func_type.lower()
+                func_type = resource_func_types.value(func_type).value
+                self.functional_types.append(func_type)
+        # else:
+        #     # TODO: Need to add all if functional type is not specified?
+        #     for func_type in resource_func_types.value:
+        #         self.functional_types.append(func_type.value)
```

Comment:
@shaileshsvc , what if component subscribe to POD event but not mentioning any functional types(data/server/control)? Do we need to allow all type of pod events notified to that component?

Reply:
The subscription entry is created by HA. It is currently hard-coded and later on will be created based on REST API subscribeForEvents when HA supports one. But still the subscription entry will be created by HA. HA can add ALL as token to mean all functional types for given resource type even in hardcoding now if needed and later when we parse REST API input parameters to create these entries.